### ISSUE TITLE
check error.message should contain key

### ISSUE THREAD

**Issue by brandonkal** [pelotom/runtypes#129]
Error messages should contain the key that is invalid.

I have programs like this. They have many properties so the default error messages are not as helpful as they could be.
```ts
const Config = rt.Record({
	domain: rt.String,
	// many more properties
})

export default function platform(cfg: rt.Static<typeof Config>) {
	Config.check(cfg)
	return JSON.stringify(cfg)
}
```
```
ValidationError: Expected string, but was undefined
```
It is not clear which property is missing.

I understand the information is available on a property, but it would be great if it was available when logging the error automatically.

This was requested in #119

**Comment by yosbelms** [comment]
@brandonkal  #119 was closed by the reporter, is a property of the error object.

```ts
try {
  Blah.check({name: 'bob'})
} catch(e) {
    console.log(e.key)
}
```

You can also use `Config.validate(cfg)` that doesn't throw.

**Comment by brandonkal** [comment]
Yes I am aware that the key is available. But it should be in the message by default as "Expected string, but was undefined" is absolutely useless on large objects.

I added it in my fork: https://github.com/brandonkal/runtypes/blob/deno/src/errors.ts and can make a PR.

**Comment by yosbelms** [comment]
@brandonkal I see

**Comment by MikeyBurkman** [comment]
To chime in on this, loggers won't know about the extra property, so that info is lost if the error is propagated to the logger without anything special wrapping it. 

I understand having _all_ the information in the message might be overkill, but to have more context would certainly be a lot more useful without forcing everyone to add boilerplate for even simple use cases.

**Comment by milanij** [comment]
@richgieg has an open PR that would solve this: https://github.com/pelotom/runtypes/pull/134

**Comment by supermacro** [comment]
This hasn't been explicitly communicated here, but another issue that I see is that `key` is for the **first** incorrect field. 

What if more than one field do not match?

Example: https://repl.it/repls/PhonySardonicMolecule

**Comment by pelotom** [comment]
I've just merged #134, does that resolve this?

> What if more than one field do not match?

There's a long-standing issue for this: https://github.com/pelotom/runtypes/issues/8. PRs would be welcome!

**Comment by yuhr** [comment]
Seems to be resolved, closing.
